### PR TITLE
[WIP] Token List Specification

### PR DESCRIPTION
**Pull request by rmeissner**
Spec for #58 

- [x] Use case scenarios
- [ ] User interfaces
- [ ] Technical details

### CODE REVIEW

File: specs/manage_tokens_list/index.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Manage Tokens - Selection List
+
+## Use Cases
+
+### Display complete token list
+As an user I want to be able to see all tokens that I have enabled and that are available.
+<details>
```
Comment (tschubotz) [inline specs/manage_tokens_list/index.md]: I like this collapsible lists thing.

File: specs/manage_tokens_list/01_main.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
+==========================================================
+Manage Tokens - Selection List
```
Comment (DmitryBespalov) [inline specs/manage_tokens_list/01_main.rst]: It seems that "selection list" is not a scenario per se, rather what I would expect to see in the scenarios:

- Display Selected Tokens
  - part of the use case on the main screen; should include handling of "default" (preselected) tokens
- Select Token
  - extension: search token
  - extension: pagination
- Deselect Token
- Update Token Information (automatic update)
  - extension: what if a previously selected token gets removed from our backend token list?

Comment (rmeissner) [reply specs/manage_tokens_list/01_main.rst]: search tokens is a different issue (#56) 
default tokens is a different issue (#57)

File: specs/manage_tokens_list/01_main.rst [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+.. contents:: Table of Contents
+    :local:
+    :depth: 2
+
+Problem Definition
+-------------------------------
+
+- User MUST be able to see all token supported by the Safe app
+- User MUST be able to enable and disable tokens that should be displayed in the asset overview 
+
```
Comment (sche) [inline specs/manage_tokens_list/01_main.rst]: What about tokens sorting?

File: specs/manage_tokens_list/01_main.rst [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+~~~~~~~~~~~~~~~~~~~~
+
+Steps
++++++
+1. The content (token list) should start loading automatically
+2. Once loading is done the tokens should be displayed
+
+Postconditions
+++++++++++++++
+- All available tokens should be displayed after the selected tokens.
```
Comment (sche) [inline specs/manage_tokens_list/01_main.rst]: This is a UX decision on how to display not selected tokens. E.g. in the current iOS version we have separate screens for selected tokens and available tokens.

File: specs/manage_tokens_list/01_main.rst [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+  * It should be possible to display an unlimited number of tokens in the list (paginated list)
```
Comment (sche) [inline specs/manage_tokens_list/01_main.rst]: IMO this is a UX decision as well. This approach has inconveniences. How the tokens should be sorted? How much should you scroll to a needed token?
I would prefer to not have pagination. 100-300 most used tokens and a search bar can work as well.

File: specs/manage_tokens_list/01_main.rst [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+
+Could not load initial token list
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+inherits from `Load token list`_
+
+Steps
++++++++
+2. An non-blocking  error should be displayed if the list could not be loaded
```
Comment (sche) [inline specs/manage_tokens_list/01_main.rst]: Currently, on iOS, we have a "retry with increasing delay" logic and do not show an error. The retrieved results and store them in db. Token list is updated on each sync. We do the tokens list sync on app start and in some other places.
We also have a merging tokens logic that takes into account "whitelisted" and "blacklisted" tokens.

File: specs/manage_tokens_list/01_main.rst (hunk 2 above)
Comment (sche) [reply specs/manage_tokens_list/01_main.rst]: I added my comments. In short, this spec is currently Android-oriented. We should discuss if we should have same ui/ux for both platforms, or different ui/ux with different spect, or more high-level user story spec with low-level technical details specs for both platforms.

**Comment by tschubotz** [comment]
Closing this since there's now a simplified spec with screen on the epic ticket.

**Pull request closed without merging**